sdk/trace/src/main/java/io/opentelemetry/sdk/trace/internal/data/ImmutableExceptionEventData.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace.internal.data;
+
+import com.google.auto.value.AutoValue;
+import com.google.auto.value.extension.memoized.Memoized;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.annotation.concurrent.Immutable;
+
+/** An effectively immutable implementation of {@link ExceptionEventData}. */
+@AutoValue
+@Immutable
+abstract class ImmutableExceptionEventData implements ExceptionEventData {
+
+  /**
+   * Returns a new immutable {@code Event}.
+   *
+   * @param epochNanos epoch timestamp in nanos of the {@code Event}.
+   * @param exception the {@link Throwable exception} of the {@code Event}.
+   * @param additionalAttributes the additional {@link Attributes} of the {@code Event}.
+   * @return a new immutable {@code Event<T>}
+   */
+  static ExceptionEventData create(
+      long epochNanos, Throwable exception, Attributes additionalAttributes) {
+
+    return new AutoValue_ImmutableExceptionEventData(epochNanos, exception, additionalAttributes);
+  }
+
+  ImmutableExceptionEventData() {}
+
+  @Override
+  public String getName() {
+    return SemanticAttributes.EXCEPTION_EVENT_NAME;
+  }
+
+  @Override
+  @Memoized
```

Comment:
Just to point out this changes the `getAttributes` for exceptions from a normal field to volatile field in the new codepath. I think it's OK just want to make sure everyone notices it

Reply:
Going for the stretch goal of having `ExceptionEventData` not generate those attributes unless the exporter tries to use them.  Assuming the exporter only calls `getAttributes` once it's probably not actually necessary to memoize them.

Reply:
If there is hesitation in deferring the generation of those attributes I can definitely back that off.  When the interface was in `tracing-incubation` I felt it was worth exploring.  The only attribute I'd prefer to avoid generating is `exception.stacktrace` as in my Spring codebases they can be quite large, but I don't think it'd a big deal to consider that later.